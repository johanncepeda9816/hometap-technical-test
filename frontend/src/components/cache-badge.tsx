export const CacheBadge = ({ isCached }: { isCached: boolean }) => (
  <span
    className={`ml-2 px-2 py-1 text-xs font-medium rounded-full ${
      isCached ? "bg-blue-100 text-blue-800" : "bg-gray-100 text-gray-800"
    }`}
  >
    {isCached ? "Cached" : "Live"}
  </span>
);
