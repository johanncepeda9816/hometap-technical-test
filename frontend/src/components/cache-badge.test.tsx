import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { CacheBadge } from "./cache-badge";
import "@testing-library/jest-dom";

describe("CacheBadge", () => {
  it('renders "Cached" with blue styling when isCached is true', () => {
    const { getByText } = render(<CacheBadge isCached={true} />);

    const badge = getByText("Cached");

    expect(badge.className).toContain("bg-blue-100");
    expect(badge.className).toContain("text-blue-800");
    expect(badge.className).toContain("ml-2");
    expect(badge.className).toContain("px-2");
    expect(badge.className).toContain("py-1");
    expect(badge.className).toContain("text-xs");
    expect(badge.className).toContain("font-medium");
    expect(badge.className).toContain("rounded-full");
  });

  it('renders "Live" with gray styling when isCached is false', () => {
    const { getByText } = render(<CacheBadge isCached={false} />);

    const badge = getByText("Live");

    expect(badge.className).toContain("bg-gray-100");
    expect(badge.className).toContain("text-gray-800");
    expect(badge.className).toContain("ml-2");
    expect(badge.className).toContain("px-2");
    expect(badge.className).toContain("py-1");
    expect(badge.className).toContain("text-xs");
    expect(badge.className).toContain("font-medium");
    expect(badge.className).toContain("rounded-full");
  });

  it("has the correct accessibility attributes", () => {
    const { getByText } = render(<CacheBadge isCached={true} />);

    const badge = getByText("Cached");

    expect(badge.tagName).toBe("SPAN");
  });

  it("snapshot test - cached state", () => {
    const { container } = render(<CacheBadge isCached={true} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("snapshot test - live state", () => {
    const { container } = render(<CacheBadge isCached={false} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
