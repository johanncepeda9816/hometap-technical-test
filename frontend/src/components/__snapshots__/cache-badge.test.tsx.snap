// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CacheBadge > snapshot test - cached state 1`] = `
<span
  class="ml-2 px-2 py-1 text-xs font-medium rounded-full bg-blue-100 text-blue-800"
>
  Cached
</span>
`;

exports[`CacheBadge > snapshot test - live state 1`] = `
<span
  class="ml-2 px-2 py-1 text-xs font-medium rounded-full bg-gray-100 text-gray-800"
>
  Live
</span>
`;
